outline/client.go

```diff
@@ -15,13 +15,43 @@
 package outline
 
 import (
+	"fmt"
+
+	internal "github.com/Jigsaw-Code/outline-go-tun2socks/outline/internal/shadowsocks"
+	"github.com/Jigsaw-Code/outline-go-tun2socks/outline/internal/utf8"
 	"github.com/Jigsaw-Code/outline-internal-sdk/transport"
 )
 
 // Client provides a transparent container for [transport.StreamDialer] and [transport.PacketListener]
 // that is exportable (as an opaque object) via gobind.
 // It's used by the connectivity test and the tun2socks handlers.
-type Client struct {
+type Client interface {
```

Comment:
If `Client` is to be a type used across packages, it perhaps needs its own package. `NewClientFromJSON` could be in another package.

Reply:
`NewClientFromJSON` is not reused, but we need some `internal.NewShadowsocksClient` to be shared.